### PR TITLE
Implement basic `/withdraw` endpoint.

### PR DESCRIPTION
**Pull request by debnil**
This PR implements the barebones of the `/withdraw` endpoint - it sets up the necessary Django pages and implements some basic tests. This composes #15 and closes [INSERT SUB-ISSUE].

### CODE REVIEW

File: src/tests/conftest.py [hunk 1]
```diff
@@ -97,6 +97,36 @@ def create_deposit_transaction():
     return create_deposit_transaction
 
 
+@pytest.fixture(scope="session")
+def acc1_usd_withdrawal_transaction_factory(usd_asset_factory):
```
Comment (msfeldstein) [inline src/tests/conftest.py]: what does the acc1 naming scheme mean?

File: src/withdraw/views.py [hunk 2]
```diff
@@ -0,0 +1,65 @@
+"""
+This module implements the logic for the `/withdraw` endpoint. This lets a user
+withdraw some asset from their Stellar account.
```
Comment (msfeldstein) [inline src/withdraw/views.py]: "This lets a user
withdraw some asset from their Stellar account into a non Stellar based asset"

File: src/withdraw/views.py [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+from rest_framework.response import Response
+
+from helpers import render_error_response
+from info.models import Asset
+from transaction.models import Transaction
+
+
+def _create_transaction_id():
+    """Creates a unique UUID for a Transaction, via checking existing entries."""
+    while True:
```
Comment (msfeldstein) [inline src/withdraw/views.py]: Do we actually need to check for uuid uniqueness?  They should always be unique right?

File: src/withdraw/views.py [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+        transaction_id = uuid.uuid4()
+        if not Transaction.objects.filter(id=transaction_id).exists():
+            break
+    return transaction_id
+
+
+def _construct_interactive_url(request, transaction_id):
+    """Constructs the URL for the `/withdraw/interactive_withdraw` page."""
```
Comment (msfeldstein) [inline src/withdraw/views.py]: mention that its for the interactive webapp not necessarily the actual url

File: src/withdraw/views.py [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+
+    # TODO: Verify optional arguments.
+
+    # Verify that the asset code exists in our database, with withdraw enabled.
+    asset = Asset.objects.filter(name=asset_code).first()
+    if not asset or not asset.withdrawal_enabled:
+        return render_error_response(f"invalid operation for asset {asset_code}")
+
+    transaction_id = _create_transaction_id()
+    url = _construct_interactive_url(request, transaction_id)
```
Comment (msfeldstein) [inline src/withdraw/views.py]: Do we need to save this transaction_id anywhere or can it be ephemeral until the user actually initiates the withdrawal?

File: src/tests/conftest.py (hunk 1 above)
Comment (debnil) [reply src/tests/conftest.py]: It just denotes the Stellar account in the `Transaction` - I'm not sure of the origin of this particular naming scheme, this was just named for consistency.

File: src/withdraw/views.py (hunk 3 above)
Comment (debnil) [reply src/withdraw/views.py]: There's a super small chance of `uuid4()` generating collisions, so we used this method just as a failsafe. (Explanation [here](https://stackoverflow.com/questions/1785503/when-should-i-use-uuid-uuid1-vs-uuid-uuid4-in-python)).

File: src/withdraw/views.py (hunk 5 above)
Comment (debnil) [reply src/withdraw/views.py]: I've just been keeping the `transaction_id` ephemeral until the user actually initiates the withdrawal. Generating a transaction whenever the top line `/withdraw` endpoint is hit probably wouldn't scale well. Plus, the `transaction_id` is preserved as a query parameter in the interactive webapp url.

**Comment by msfeldstein** [comment]
Lgtm

**Pull request merged**